src/search/landmarks/exploration.cc

```diff
@@ -147,124 +122,91 @@ void Exploration::build_unary_operators(const OperatorProxy &op) {
 */
 void Exploration::setup_exploration_queue(
     const State &state, const vector<FactPair> &excluded_props,
-    const unordered_set<int> &excluded_op_ids, bool use_h_max) {
+    const unordered_set<int> &excluded_op_ids) {
     prop_queue.clear();
 
     for (size_t var_id = 0; var_id < propositions.size(); ++var_id) {
         for (size_t value = 0; value < propositions[var_id].size(); ++value) {
             ExProposition &prop = propositions[var_id][value];
-            prop.h_add_cost = -1;
+            prop.excluded = false;
             prop.h_max_cost = -1;
-            prop.depth = -1;
-            prop.marked = false;
         }
     }
 
     for (const FactPair &fact : excluded_props) {
         ExProposition &prop = propositions[fact.var][fact.value];
-        prop.h_add_cost = -2;
+        prop.excluded = true;
     }
 
     // Deal with current state.
     for (FactProxy fact : state) {
         ExProposition *init_prop = &propositions[fact.get_variable().get_id()][fact.get_value()];
-        enqueue_if_necessary(init_prop, 0, 0, 0, use_h_max);
+        enqueue_if_necessary(init_prop, 0, 0);
     }
 
     // Initialize operator data, deal with precondition-free operators/axioms.
     for (ExUnaryOperator &op : unary_operators) {
+        op.excluded = false; // Reset from previous exploration
```

Comment:
It was needed in the old code, but happened only after the for-loop (it was line 182 before we started: `op.h_add_cost = op.base_cost`). The reason it moved is probably because we removed it completely in a first approach and only later realized that it must be reset whenever we start a new exploration.

Reply:
I see, thanks. I think that logic is a bit convoluted, both the split initialization where we already essentially have an if/else branch for excluded/non-excluded actions below anyway, and also the fact that the logic below uses `continue` rather than s normal `if/else`. Would be good to clean this up further, but no need for further changes now.